src/Microsoft.VisualStudio.ProjectSystem.Managed.TestServices/ProjectSystem/ProjectTreePropertiesProviderExtensions.cs

```diff
@@ -15,13 +16,23 @@ internal static class ProjectTreePropertiesProviderExtensions
         /// </summary>
         public static IProjectTree ChangePropertyValuesForEntireTree(this IProjectTreePropertiesProvider propertiesProvider, IProjectTree tree)
```

Comment:
If we are not returning the same instance that was initially provided then the method name should reflect that we are cloning the given object and not mutating the object. May be `CreateEntireTreeWithChangedPropertyValues`? or maybe something else which captures the essence of what is happening here?